R/util.R

```diff
@@ -0,0 +1,279 @@
+              suffix=suffixes)
+  
+  return(combined_results)
+}
+
+#' Create an interactive scatter plot comparing two DQD results
+#' 
+#' Matches checks from the two provided DQD results and saves two outputs:
+#' \itemize{
+#'  \item a table with all checks that are different
+#'  \item an interactive plot (html) with old fail percentage versus new fail
+#'  percentage
+#' } 
+#' This makes it easy to identify differences between subsequent runs of DQD.
+#' Nothing will be created if there are no differences between the given DQD results.
+#' See also: \link[TBD]{TBD - link to abstract}
+#' 
+#' @param jsonPath.old the path to the old DQD json results file
+#' @param jsonPath.new the path to the new DQD json results file
+#' @param savingDir (optional) the path to the folder where the output should be written.
+#'                  if not given, no output will be written.
+#' 
+#' @author Elena Garcia Lara
+#' @author Maxim Moinat
+#' 
+#' @return An interactive plotly figure or nothing if no differences are found.
+#' @export
+#' 
+#' @examples 
+#' \dontrun{
+#'   plotCompareDqdResults("dqd_results_1.json", "dqd_results_2.json", "output")
+#' }
+plotCompareDqdResults <- function(jsonPath.old, jsonPath.new, savingDir = NA){
+  combinedResult <- .joinDqdResults(jsonPath.old, jsonPath.new, suffixes = c(".old", ".new"))
+  
+  # Only keep changed
+  combinedResult <- combinedResult %>%
+                      filter(PCT_VIOLATED_ROWS.old != PCT_VIOLATED_ROWS.new)
+  
+  # When no difference found, exit function
+  if(nrow(combinedResult)==0){
+    stop("No differences found.")
+  }
+  
+  # Visualization
+  p <- combinedResult %>%
+    mutate(       
+      fail_status = ifelse(FAILED.old, 
+                           ifelse(FAILED.new, "Fail→Fail", "Fail→Pass"),
+                           ifelse(FAILED.new, "Pass→Fail", "Pass→Pass")
+                           ),
+      pct_old = round(PCT_VIOLATED_ROWS.old*100, digits=2),
+      pct_new = round(PCT_VIOLATED_ROWS.new*100, digits=2)
+    ) %>% 
+    ggplot(aes(x=pct_old,
+               y=pct_new,
+               colour=fail_status,
+               text=paste(
+                 sprintf('<br><i>Check name: </i>%s', CHECK_NAME),
+                 sprintf('<br><i>Table: </i>%s', CDM_TABLE_NAME),
+                 sprintf('<br><i>Field: </i>%s', CDM_FIELD_NAME),
+                 sprintf('<br><i>Threshold value: </i>%.1f%%', THRESHOLD_VALUE.new),
+                 sprintf('<br><b><i>old: </i> %.2f%% </b>', pct_old),
+                 sprintf('<br><b><i>new: </i> %.2f%% </b>', pct_new)
+               ), alpha=0.6)) +
+    geom_point() +
+    geom_abline(colour="gray", linetype = "dashed")+
+    scale_colour_manual(labels = c("Fail→Pass", "Fail→Fail", "Pass→Pass", "Pass→Fail"),
+                        values = c("Pass→Pass" = "lightblue", "Fail→Fail" = "chocolate1",
+                                   "Fail→Pass" = "darkblue", "Pass→Fail" = "coral"))+
```

Comment:
```suggestion
    scale_colour_manual(labels = c("Fail-to-Pass", "Fail-to-Fail", "Pass-to-Pass", "Pass-to-Fail"),
                        values = c("Pass-to-Pass" = "lightblue", "Fail-to-Fail" = "chocolate1",
                                   "Fail-to-Pass" = "darkblue", "Pass-to-Fail" = "coral"))+
```

Same as before :(
Love the colour choice :)